src/ReinforcementLearning/SimpleAgent/Agent.h

```diff
@@ -0,0 +1,89 @@
+#ifndef AGENT_H
+#define AGENT_H
+
+#include "RandomPolicy.h"
+#include "Policy.h"
+#include "../../alignment/State.h"
+#include <stdlib.h>
+#include <random>
+#include <chrono>
+#include <utility>
+#include <iostream>
+#include <map>
+#include "../../alignment/Graph.h"
+#include "Episode.h"
+
+using std::vector;
+
+/**TODO: Include score (maybe expected score) in history.
+ *       The policy is supposed to map states to a vector of action probabilities in my view.
+ */
+
+/** \brief This Agent class selects edges according to a policy.
+*/
+class Agent {
+    public:
+    state* constState;
+    Policy* policy;
+    Graph graph;
+    Agent(){};
+    Agent(state* cs, Graph& g, Policy* pol){
+        constState = cs;
+        policy = pol;
```

Comment:
State and policy can be members of Agent instead of pointers to something that is created outside. For policymorphic objects like policy, you can use `std::unique_ptr<Policy>` (https://en.cppreference.com/w/cpp/memory/unique_ptr). Maybe initialize the Agent with an enum {random, learned_policy} to indicate which policy should be used.